modin/pandas/base.py

```diff
@@ -3111,21 +3111,49 @@ def __init__(
         axis = self._dataframe._get_axis_number(axis)
         # FIXME: this should be converted into a dict to ensure simplicity
         # of handling resample parameters at the query compiler level.
-        self.resample_args = [
-            rule,
-            axis,
-            closed,
-            label,
-            convention,
-            kind,
-            loffset,
-            base,
-            on,
-            level,
-            origin,
-            offset,
-        ]
-        self.__groups = self.__get_groups(*self.resample_args)
+        self.resample_args = {
+            "rule": rule,
+            "axis": axis,
+            "closed": closed,
+            "label": label,
+            "convention": convention,
+            "kind": kind,
+            "loffset": loffset,
+            "base": base,
+            "on": on,
+            "level": level,
+            "origin": origin,
+            "offset": offset,
+        }
+        self.__groups = self.__get_groups(*self.resample_args.values())
+
+    def __getitem__(self, key):
+        if len(self._dataframe.columns.intersection(key)) != len(key):
+            missed_keys = list(set(key).difference(self._dataframe.columns))
+            raise KeyError(f"Columns {str(missed_keys)[1:-1]} don't exist.")
+        return self._gotitem(list(key))
+
+    def _gotitem(self, key):
```

Comment:
Possible, it will be better to move implementation of this function in `__getitem__` method because this method doesn't have any complex logic and it is only used  in __getitem__.

Reply:
fixed